opal/mca/threads/pthreads/threads_pthreads_yield.c

```diff
@@ -0,0 +1,54 @@
+/* -*- Mode: C; c-basic-offset:4 ; indent-tabs-mode:nil -*- */
+/*
+ * Copyright (c) 2020      High Performance Computing Center Stuttgart,
+ *                         University of Stuttgart.  All rights reserved.
+ *
+ * $COPYRIGHT$
+ *
+ * Additional copyrights may follow
+ *
+ * $HEADER$
+ */
+
+#include <unistd.h>
+#include <time.h>
+#ifdef HAVE_SCHED_H
+#include <sched.h>
+#endif
+
+#include "opal/mca/threads/threads.h"
+#include "opal/mca/threads/pthreads/threads_pthreads.h"
+#include "opal/sys/atomic.h"
+
+
+opal_threads_pthreads_yield_strategy_t opal_threads_pthreads_yield_strategy = OPAL_PTHREADS_YIELD_SCHED_YIELD;
+uint64_t opal_threads_pthreads_yield_usleep = 1;
+uint64_t opal_threads_pthreads_yield_nsleep = 100;
+
+
+void opal_thread_yield(void)
+{
+    switch (opal_threads_pthreads_yield_strategy) {
+    case OPAL_PTHREADS_YIELD_SCHED_YIELD:
+#ifdef HAVE_SCHED_H
+        sched_yield();
+        break;
+#endif
+        /* fall-through if sched_yield is not available */
+    case OPAL_PTHREADS_YIELD_USLEEP:
+        usleep(opal_threads_pthreads_yield_usleep);
+        break;
+    case OPAL_PTHREADS_YIELD_NANOSLEEP:
+    {
+        static struct timespec sleeptime = {.tv_sec = -1, .tv_nsec = -1};
```

Comment:
There is little potential for race conditions here, but the potential is inside. You have added a registration function, why don't you compute this value there?

Reply:
Agreed, it's now set up after reading the mca parameter.